bin/check-smart.rb

```diff
@@ -78,8 +78,17 @@ def device_path
   #
   def check_smart_capability!
     output = `sudo #{@smart_binary} -i #{device_path}`
+
+    # Newer smartctl
     @smart_available = !output.scan(/SMART support is:\s+Available/).empty?
     @smart_enabled = !output.scan(/SMART support is:\s+Enabled/).empty?
+
+    unless smart_capable?
```

Comment:
I think this can be made much cleaner IMHO like this:
```ruby
if smart_capable?
  # Newer smartctl
   @smart_available = !output.scan(/SMART support is:\s+Available/).empty?
   @smart_enabled = !output.scan(/SMART support is:\s+Enabled/).empty?
else
  # Older smartctl
  @smart_available = !output.scan(/Device supports SMART/).empty?
  @smart_enabled = !output.scan(/and is Enabled/).empty?
end
```

Reply:
The issue here is that the initial
```
if smart_capable?
```
will always be false as up to this point the disk has not been assessed at all.
The logic flow needs to be basically
1. Look for enabled/available first off by checking against the new output.
2. If the device still reports as not being smart capable, it may be that its the old out put, test again against the old output.

Reply:
I see, that's what happens when I don't review it well enough.